### PR TITLE
MISUV-7940-model-unit-tests

### PR DESCRIPTION
**Pull request by james-king-hmrc**
model-unit-tests

### CODE REVIEW

File: test/connectors/itsastatus/OptOutUpdateRequestModelSpec.scala [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+}
+
+object OptOutUpdateRequestModelSpec {
+
+  private val requestObj = OptOutUpdateRequest("2023-2024", optOutUpdateReason)
+  private val requestJson: JsValue = Json.obj(
+    "taxYear" -> "2023-2024",
+    "updateReason" -> "10",
+  )
```
Comment (ian-hmrc) [inline test/connectors/itsastatus/OptOutUpdateRequestModelSpec.scala:63]: I think what we're trying to achieve with this test is to show that we can parse what we're expecting from the API, and can copy and paste the examples from the API spec into this test. For that reason, I like to parse the JSON from a string that can be copied 'n' pasted in. Here's an example I've got lying around...
```scala
    "parse from json Mandated" in {
      Json.parse(
        """{
        "submittedOn": "2024-01-10T06:14:00Z",
        "status": "MTD Mandated",
        "statusReason": "Sign up - no return available"
      }""").validate[StatusDetail] mustBe(JsSuccess
        (StatusDetail(
          submittedOn = "2024-01-10T06:14:00Z",
          status = ITSAStatus.Mandated,
          statusReason = "Sign up - no return available",
          businessIncomePriorTo2Years = None)))
    }
```

File: test/connectors/itsastatus/OptOutUpdateRequestModelSpec.scala [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import play.api.libs.json.{JsSuccess, JsValue, Json}
+
+class OptOutUpdateRequestModelSpec extends AnyWordSpec with Matchers {
+
+  "the request model" should {
+    "read from json" in {
+      Json.fromJson[OptOutUpdateRequest](requestJson) shouldBe JsSuccess(requestObj)
+    }
```
[the remark this answers is not in the document]
Comment (james-king-hmrc) [reply test/connectors/itsastatus/OptOutUpdateRequestModelSpec.scala]: done

File: test/connectors/itsastatus/OptOutUpdateRequestModelSpec.scala (hunk 1 above)
Comment (james-king-hmrc) [reply test/connectors/itsastatus/OptOutUpdateRequestModelSpec.scala]: done

File: test/connectors/itsastatus/OptOutUpdateRequestModelSpec.scala [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    "write to json" in {
+      Json.toJson(requestObj) shouldBe requestJson
+    }
```
[the remark this answers is not in the document]
Comment (james-king-hmrc) [reply test/connectors/itsastatus/OptOutUpdateRequestModelSpec.scala:38]: done

File: test/connectors/itsastatus/OptOutUpdateRequestModelSpec.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (ian-hmrc) [reply test/connectors/itsastatus/OptOutUpdateRequestModelSpec.scala]: Cool

File: test/connectors/itsastatus/OptOutUpdateRequestModelSpec.scala (hunk 1 above)
Comment (ian-hmrc) [reply test/connectors/itsastatus/OptOutUpdateRequestModelSpec.scala]: Cool

File: test/connectors/itsastatus/OptOutUpdateRequestModelSpec.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (ian-hmrc) [reply test/connectors/itsastatus/OptOutUpdateRequestModelSpec.scala:38]: I think this is not quite right. The request needs to be written to json to be sent. The ones below for reading the response are good though.

[the remark this answers is not in the document]
Comment (james-king-hmrc) [reply test/connectors/itsastatus/OptOutUpdateRequestModelSpec.scala:38]: done

[the remark this answers is not in the document]
Comment (ian-hmrc) [reply test/connectors/itsastatus/OptOutUpdateRequestModelSpec.scala:38]: Cool. :)

**Review by ian-hmrc — APPROVED** [review]
LGTM

**Pull request merged**